From 82e7c8bcebb0c3bdcf27487fc770c76bff1444c6 Mon Sep 17 00:00:00 2001
From: Cedric Poignet <cedric.poignet@intel.com>
Date: Mon, 3 Jun 2013 11:12:32 +0200
Subject: [PORT FROM MAIN][BT] Remove Bluetooth if BOARD_HAVE_BLUETOOTH is false

BZ: 110380

In case the Bluetooth feature is not functional or if the hardware
is not ready, the BT application can crash.

Setting the BOARD_HAVE_BLUETOOTH flag to false helps us to disable BT
support for a given platform with minimmum side effects.

Change-Id: Ib7e8ff22045cf62ef3280ba538d3997bcf2f2611
Orig-Change-Id: Iaac337860ea788c25303918ae745ce7a57c30b42
Category: device enablement
Domain: CWS.BT-Common
Origin: internal
Upstream-Candidate: no, not-aosp
Signed-off-by: Cedric Poignet <cedric.poignet@intel.com>
Reviewed-on: http://android.intel.com:8080/111520
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Seibel, Eric <eric.seibel@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 target/product/generic_no_telephony.mk |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/target/product/generic_no_telephony.mk b/target/product/generic_no_telephony.mk
index 1155b4e..1622158 100644
--- a/target/product/generic_no_telephony.mk
+++ b/target/product/generic_no_telephony.mk
@@ -21,7 +21,6 @@ PRODUCT_POLICY := android.policy_phone
 
 PRODUCT_PACKAGES := \
     DeskClock \
-    Bluetooth \
     Calculator \
     Calendar \
     Camera2 \
@@ -44,11 +43,16 @@ PRODUCT_PACKAGES := \
     SystemUI \
     TeleService \
     CalendarProvider \
-    bluetooth-health \
     hostapd \
     wpa_supplicant.conf \
     WallpaperCropper
 
+ifneq ($(BOARD_HAVE_BLUETOOTH),false)
+PRODUCT_PACKAGES += \
+    Bluetooth \
+    bluetooth-health
+endif
+
 PRODUCT_PACKAGES += \
     audio \
     clatd \
-- 
1.7.4.1

